databricks_cli/stack/api.py

```diff
@@ -234,6 +245,58 @@ def _update_job(self, job_settings, job_id):
         click.echo("Updating Job")
         self.jobs_client.reset_job({'job_id': job_id, 'new_settings': job_settings})
 
+    def _deploy_workspace(self, resource_properties, physical_id, overwrite):
+        """
+        Deploy workspace asset.
+
+        :param resource_properties: dict of properties for the workspace asset. Must contain the
+        'source_path' and 'path' fields. The other fields will be inferred if not provided.
+        :param physical_id: dict containing physical identifier of workspace asset on databricks.
+        Should contain the field 'path'.
+        :param overwrite: Whether or not to overwrite the contents of workspace notebooks.
+        :return: (dict, dict) of (physical_id, deploy_output). physical_id is the physical ID for
+        the stack status that contains the workspace path of the notebook or directory on datbricks.
+        deploy_output. Is the initial information about the asset on databricks at deploy time.
+        """
+        try:
+            local_path = resource_properties['source_path']
+            workspace_path = resource_properties['path']
+        except KeyError as e:
+            raise StackError("%s doesn't exist in workspace resource properties" % str(e))
+
+        lang_fmt = WorkspaceLanguage.to_language_and_format(local_path)  # Guess language and format
+        language, fmt = None, None
+        if lang_fmt:
+            language, fmt = lang_fmt
```

Comment:
Here we should just simply do
```
(language, fmt) =  WorkspaceLanguage.to_language_and_format(local_path)
```
And throw if we see any inconsistency. We do not want to resolve conflicts here because it is very hard to predict which one is the truth when the file name and the language field is not consistent.

Reply:
Actually, if everything is encoded in the path, then we should just remove "language" and "format" from the jsonnet config entirely.

Reply:
for language, some users may not have a file extension on their files and it may not be appropriate to force them to put file extensions since on databricks, file extensions are not put on by default. Not so sure about format though. I can test out deploying/downloading as an HTML format or some other format

Reply:
Also, doing `(language, fmt) = WorkspaceLanguage.to_language_and_format(local_path)` returns only `None` when`to_language_and_format` doesn't detect a language or format, that's why i used the `lang_fmt` variable.

Reply:
I plan to keep language and format in, while raising an error if there are conflicts.